x/version/README.md

```diff
@@ -0,0 +1,106 @@
+# `x/version`
+
+## Abstract
+
+The version module will increment the app version in the header per a predefined
+mapping of app versions to heights. This allows for application logic to be
+routed in the same binary, which enables single binary syncs and upgrades.
+
+It accomplishes this by wrapping the `BaseApp`'s `EndBlock` method and checking
+if the current height is equal to the height at which the app version should be
+incremented. If it is, then the app version is incremented and the new version
+is set in the header.
+
+```go
+func (app *App) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBlock) {
+	res = app.BaseApp.EndBlock(req)
+	ctx := app.GetContextForDeliverTx([]byte{1})
+	return appversion.EndBlocker(ctx, app.VersionKeeper, res)
+}
+
+// EndBlocker will modify the app version if the current height is equal to
+// a predefined height at which the app version should be changed.
+func EndBlocker(ctx sdk.Context, keeper Keeper, resp abci.ResponseEndBlock) abci.ResponseEndBlock {
+	newAppVersion := keeper.GetVersion(ctx)
+	if ctx.BlockHeader().Version.App != newAppVersion {
+		resp.ConsensusParamUpdates.Version = &coretypes.VersionParams{
+			AppVersion: newAppVersion,
+		}
+		// set the version in the application to ensure that tendermint is
+		// passed the correct value upon rebooting
+		keeper.versionSetter.SetProtocolVersion(newAppVersion)
+	}
+	return resp
+}
+```
+
+## State
+
+The state in the version module is abnormal because it can be modified by the
+party building the binary without breaking the app hash. This is because the
+state is not stored in the iavl tree, but rather as a simple predefined golang
+map. It's important to note that even though this state is not in the state
```

Comment:
Can you add a sentence where this map is persisted if not in the state? (even if that is explained in more detail below)